sources/rules.h

```diff
@@ -122,8 +121,31 @@ struct od_rule
 	int                     client_max;
 	int                     log_debug;
 	od_list_t               link;
+    /* pause/resume related */
+    pthread_mutex_t lock;
```

Comment:
I've studied how rules are updated. So I implemented setting pause/resume state the way that circumvent race conditions while config reload is performed.

I don't like the idea of putting state into route because route doesn't represent a DB. Storing state inside a rule isn't good to. The best solution would be to create a new entity that will represent DB's overall state and contain both pause/resume-state and rule(it looks like an atomic part of state). But this is a big change to architecture and should be discussed. Who should I discuss it with? We can discuss it here or I can meet @x4m .

Reply:
I see no reason for this extension regarding only this change. JFYI statistics is also calculated per-database, you should follow the same logic here I believe.

Reply:
It is inconvenient to put pause/resume state into routes. If there is a rule with no routes referencing it pause would make no effect. The best user can get is "no such storage" error but there is such storage. We just can't store pause/resume this way safely.

Reply:
@x4m and me discussed that and he approved the idea of adding mutable state object.

Reply:
I do not see a good reason why you need to implement it, current implementation is optimal for router and other subsystems because you have to support rules where you create automatic routes for `all_database`:`user`, `database`:`all_users`, `all_databases:all_users` as well. It will be harder to maintain it in general and keep up dependencies.

Yet if you brave enough to do it correctly, I would like to see it as a separate PR.